Assets/BossRoom/Scripts/Shared/Game/State/GameStateBehaviour.cs

```diff
@@ -87,8 +105,14 @@ protected virtual void Start()
             }
         }
 
+        protected virtual void InitializeScope()
+        {
+            m_Scope = new DIScope(DIScope.RootScope);
```

Comment:
I would move this line to line 65, before we call this method - this will be where people need to actually bind the dependencies. Otherwise someone might forget to create the m_Scope ?

Reply:
I expect that in this case we would get errors when running if we don't properly override this while calling base.InitializeScope. But it could be moved to Awake instead and make this method empty for the parent class. That would be one less thing for users to worry about when creating new states. We would still need to make sure that base.awake and base.start are called, but that would be it.